src/test/java/io/gingersnapproject/testcontainers/BaseGingersnapResourceLifecycleManager.java

```diff
@@ -31,11 +31,13 @@
 
 public class BaseGingersnapResourceLifecycleManager implements
       QuarkusTestResourceConfigurableLifecycleManager<WithDatabase>, QuarkusTestBeforeClassCallback {
+
+   static final Map<String, BaseGingersnapResourceLifecycleManager> TESTING_RESOURCES = new ConcurrentHashMap<>();
```

Comment:
Why do we need a rule name as a key here when we always set to the value to `this`?

If this isn't needed we can take advantage of the fact we're implementing `QuarkusTestResourceConfigurableLifecycleManager` and just inject the `HotRodContainer` and `JdbcDatabaseContainer` directly into the tests that require them:

https://quarkus.io/guides/getting-started-testing#altering-the-test-class

Reply:
Nice, let me try that. Initially, this was static because there were two instances, one by Quarkus, which is the one that starts the containers, and another created by JUnit to use the "beforeAll" callback.

Reply:
That worked perfectly, way better than using the static variable. I tried to do something for the latch map but had no success. I can look into it in the future again, but for now, it should work.